src/core/pointcloud/qgspointcloudblockrequest.h

```diff
@@ -0,0 +1,75 @@
+/***************************************************************************
+                         qgspointcloudblockrequest.h
+                         --------------------
+    begin                : March 2021
+    copyright            : (C) 2021 by Belgacem Nedjima
+    email                : belgacem dot nedjima at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSPOINTCLOUDBLOCKREQUEST_H
+#define QGSPOINTCLOUDBLOCKREQUEST_H
+
+#include <QObject>
+
+#include "qgspointcloudattribute.h"
+#include "qgstiledownloadmanager.h"
+#include "qgspointcloudindex.h"
+
+#define SIP_NO_FILE
+
+class QgsPointCloudAttributeCollection;
+class QgsPointCloudBlock;
+
+/**
+ * \ingroup core
+ * \brief Base class for handling loading QgsPointCloudBlock asynchronously
+ *
+ * \note The API is considered EXPERIMENTAL and can be changed without a notice
+ *
+ * \since QGIS 3.20
+ */
+class CORE_EXPORT QgsPointCloudBlockRequest : public QObject
+{
+    Q_OBJECT
+  public:
+
+    /**
+     * QgsPointCloudBlockRequest constructor
+     * Note: It is the responsablitiy of the caller to delete the block if it was loaded correctly
+     */
+    QgsPointCloudBlockRequest( const IndexedPointCloudNode &node, const QString &Uri, const QString &dataType, const QgsPointCloudAttributeCollection &attributes, const QgsPointCloudAttributeCollection &requestedAttributes );
+
+    /**
+     * Returns the requested block. if the returned block is nullptr, that means the data request failed
+     * Note: It is the responsablitiy of the caller to delete the block if it was loaded correctly
```

Comment:
Why switching back to giving the responsibility of the caller to delete the block? :thinking:

Reply:
In QgsRemoteEptPointCloudIndex::nodeData we internally use QgsPointCloudBlockRequest and take QgsPointCloudBlock from it and return the block to the caller, so depending on the situation we don't want QgsPointCloudBlock to be owned by QgsPointCloudBlockRequest.